designs/shelly/src/raglansleeve.mjs

```diff
@@ -0,0 +1,306 @@
+      from: points.frontArmholeScooped,
+      to: points.backArmholeScooped,
+      y: points.frontSleeve.y + (30 + sa),
+    })
+
+    macro('vd', {
+      from: points.backNeck,
+      to: points.backArmholeScoopEnd,
+      x: points.backArmholeScooped.x + (15 + sa),
+    })
+    macro('vd', {
+      from: points.backArmholeScoopEnd,
+      to: points.backArmholeScooped,
+      x: points.backArmholeScooped.x + (15 + sa),
+    })
+    macro('vd', {
+      from: points.backNeck,
+      to: points.backArmholeScooped,
+      x: points.backArmholeScooped.x + (30 + sa),
+    })
+    macro('vd', {
+      from: points.backArmholeScooped,
+      to: points.backSleeve,
+      x: points.backArmholeScooped.x + (15 + sa),
+    })
+    macro('vd', {
+      from: points.backNeck,
+      to: points.backSleeve,
+      x: points.backArmholeScooped.x + (45 + sa),
+    })
+
+    macro('hd', {
+      from: points.frontArmholeScooped,
+      to: points.frontArmholeScoopEnd,
+      y: Math.max(points.frontNeck.y, points.backNeck.y) - (15 + sa),
+    })
+    macro('hd', {
+      from: points.frontArmholeScoopEnd,
+      to: points.frontNeck,
+      y: Math.max(points.frontNeck.y, points.backNeck.y) - (15 + sa),
+    })
+    macro('hd', {
+      from: points.frontNeck,
+      to: points.backNeck,
+      y: Math.max(points.frontNeck.y, points.backNeck.y) - (15 + sa),
+    })
+    macro('hd', {
+      from: points.backNeck,
+      to: points.backArmholeScoopEnd,
+      y: Math.max(points.frontNeck.y, points.backNeck.y) - (15 + sa),
+    })
+    macro('hd', {
+      from: points.backArmholeScoopEnd,
+      to: points.backArmholeScooped,
+      y: Math.max(points.frontNeck.y, points.backNeck.y) - (15 + sa),
+    })
+    macro('hd', {
+      from: points.backNeck,
+      to: points.backArmholeScooped,
+      y: Math.max(points.frontNeck.y, points.backNeck.y) - (30 + sa),
+    })
+    macro('hd', {
+      from: points.frontArmholeScooped,
+      to: points.frontNeck,
+      y: Math.max(points.frontNeck.y, points.backNeck.y) - (30 + sa),
+    })
+    macro('hd', {
+      from: points.frontArmholeScooped,
+      to: points.backArmholeScooped,
+      y: Math.max(points.frontNeck.y, points.backNeck.y) - (45 + sa),
+    })
+  }
+
+  if (complete) {
+    points.grainlineBottom = points.backSleeve.shiftFractionTowards(points.frontSleeve, 0.5)
+    points.grainlineTop = points.raglanCenter.shift(270, neckRadius)
+
+    macro('grainline', {
+      from: points.grainlineTop,
+      to: points.grainlineBottom,
+    })
+
+    // notches
+    snippets.frontArmholeScoopEnd = new Snippet('notch', points.frontArmholeScoopEnd)
+    snippets.backArmholeScoopEnd = new Snippet('bnotch', points.backArmholeScoopEnd)
+
+    points.title = new Point(0, points.backSleeve.y / 3)
+    macro('title', { at: points.title, nr: 3, title: 'sleeve' })
+
+    points.logo = points.title.shift(-90, 70)
+    snippets.logo = new Snippet('logo', points.logo)
+    points.scalebox = points.logo.shift(-90, 70)
```

Comment:
Again, consider avoiding using hardcoded "70 mm) for locating the logo and scalebox.

Reply:
That's copy and pasted from an existing pattern (I don't remember which), so there may be some fixes needed to that pattern. I'll take a look at the code and figure out how best to fix it).

Reply:
Given that the logo itself doesn't scale, maybe it would be better to check a key measurement (sleeve length for the sleeve, body length for the body) and only draw the logo if there's room?

Reply:
The logo will scale once PR #3999 is in.

Reply:
It should be fixed now, using the scale factor.